File: Difficulty/Easy/21. Merge Two Sorted Lists/Solutions/Solution.java
class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int data) {
        this.val = data;
    }
}

class Solution {
    public static ListNode solution(ListNode list1, ListNode list2) {
        ListNode lst, temp;
        temp = new ListNode();
        lst = temp;
        while (list1 != null && list2 != null) {
            if (list1.val < list2.val) {
                temp.next = list1;
                list1 = list1.next;
            } else {
                temp.next = list2;
                list2 = list2.next;
            }
            temp = temp.next;
        }
        if (list1 != null) {
            temp.next = list1;
        } else {
            temp.next = list2;
        }

        return lst.next;
    }
}

File: Difficulty/Easy/21. Merge Two Sorted Lists/Solutions/Solution.py
class ListNode:
    def __init__(self, data):
        self.data = data
        self.next = None
        
class Solution:
    def solution(list1 : ListNode, list2 : ListNode) -> ListNode:
        lst = temp = ListNode(0)
        
        while list1 and list2:
            if list1.data < list2.data:
                temp.next = list1
                list1 = list1.next
            else:
                temp.next = list2
                list2 = list2.next
            temp = temp.next
        if list1:
            temp.next = list1
        if list2:
            temp.next = list2
        return lst.next